Branch/BranchQRCode.m

```diff
@@ -0,0 +1,211 @@
+//
+//  BranchQRCode.m
+//  BranchMacOS
+//
+//  Created by Nipun Singh on 5/23/22.
+//  Copyright © 2022 Branch, Inc. All rights reserved.
+//
+
+#import <LinkPresentation/LPLinkMetadata.h>
+#import "BranchQRCode.h"
+#import "Branch.h"
+#import "BNCQRCodeCache.h"
+
+@implementation BranchQRCode
+
+NSString *buoTitle;
+CIImage *qrCodeImage;
+
+- (void) setMargin:(NSNumber *)margin {
+    if (margin.intValue > 20) {
+        margin = @(20);
+        BNCLogWarning(@"Margin was reduced to the maximum of 20.");
+    }
+    if (margin.intValue < 1) {
```

Comment:
Check should be  if (margin.intValue < 0) {

Reply:
Updated the comment to reflect the minimum.